cli/io/cbt.py

```diff
@@ -0,0 +1,189 @@
+import copy
+import json
+import re
+from pathlib import Path
+
+import requests
+import yaml
+
+from cli import Cli
+from utility.log import Log
+
+required_packages = [
+    "blktrace",
+    "seekwatcher",
+    "perf",
+    "valgrind",
+    "fio",
+    "cosbench",
+    "pdsh",
+]
+log = Log(__name__)
+
+
+class CbtError(Exception):
+    pass
+
+
+class CBT(Cli):
+    def __init__(self, primary_client):
+        super(CBT, self).__init__(primary_client)
+        self.primary_client = primary_client
+
+        self.required_packages = [
+            "blktrace",
+            "seekwatcher",
+            "perf",
+            "valgrind",
+            "fio",
+            "cosbench",
+            "pdsh",
+            "pdsh-rcmd-ssh",
+        ]
+        self.cbt_conf = """cluster:
+          user: 'cephuser'
+          head: "{head_placeholder}"
+          clients: ["{client_placeholder}"]
+          osds: {osd_nodes}
+          mons:
+            {mon_placeholder}:
+              a: "{ip_placeholder}:{port_placeholder}"
+          osds_per_node: 1
+          fs: 'xfs'
+          mkfs_opts: '-f -i size=2048'
+          mount_opts: '-o inode64,noatime,logbsize=256k'
+          conf_file: '/etc/ceph/ceph.conf'
+          rbd_cmd: 'sudo /usr/bin/rbd'
+          ceph_cmd: 'sudo /usr/bin/ceph'
+          iterations: 1
+          use_existing: True
+          clusterid: "ceph"
+          tmp_dir: "/tmp/cbt"
+          pool_profiles:
+            rbd:
+              pg_size: 256
+              pgp_size: 256
+              replication: 3
+        """
+
+    def setup_CBT(self):
+        self.execute(cmd="git clone https://github.com/ceph/cbt.git")
+        self.execute(cmd="yum install ceph-common -y")
+        self.enable_epel()
+        for package in self.required_packages:
+            self.execute(cmd=f"yum install {package} -y")
+        out, rc = self.execute(
+            cmd="python3 -m venv .venv;.venv/bin/python -m pip install --upgrade pip;"
+            ".venv/bin/python -m pip install -r cbt/requirements.txt",
+        )
```

Comment:
IMO, we should avoid multiple calls. Opt to use a script to enable a single remote execution. This would reduce the number of failure points.